File: 6588.py
import sys

primes = [True if i%2 == 1 else False for i in range(1000001)]
primes[0] = False
primes[1] = False

for i in range(3, 1000000, 2):
    if primes[i]:
        j = 2
        while (i*j) < 1000001:
            primes[i*j] = False
            j+=1
while True:
    n = int(sys.stdin.readline())

    if n == 0:
        break

    for p in range(n-3, 2, -2):
        if primes[p] and primes[n-p]:
            print(str(n)+' = '+str(n-p)+' + '+str(p))
            break
    else:
        print("Goldbach's conjecture is wrong.")

File: 6588.c
#include<stdio.h>

int isPrime(int n){
	int i;
	for(i=3;i<n;i+=2){
		if(n%i == 0){
			return 0;
		}
	}
	return 1;
}
int main(){
	while(1){
		int n = 0;
		scanf("%d", &n);
		
		if(n==0){
			break;
		}
		int i, flag = 1;
		for(i=n-3;i>=3;i-=2){
			if(isPrime(i) && isPrime(n-i)){
				printf("%d = %d + %d\n", n, n-i, i);
				flag = 0;
				break;
			}
		}
		if(flag){
			printf("Goldbach's conjecture is wrong.\n");
		}
	}
}
